scene/gui/control.cpp

```diff
@@ -2358,6 +2358,25 @@ void Control::_window_find_focus_neighbor(const Vector2 &p_dir, Node *p_at, cons
 		points[2] = xform.xform(c->get_size());
 		points[3] = xform.xform(Point2(0, c->get_size().y));
 
+		// Tie-breaking aims to address situations where a potential focus neighbor's bounding rect
+		// is right next to the currently focused control (e.g. in BoxContainer with
+		// separation overridden to 0). This needs specific handling so that the correct
+		// focus neighbor is selected.
+
+		// Calculate centers of the potential neighbor, currently focused, and closest controls.
+		Point2 center = xform.xform(0.5 * c->get_size());
+		// We only have the points, not an actual reference.
+		Point2 p_center = 0.25 * (p_points[0] + p_points[1] + p_points[2] + p_points[3]);
+		Point2 closest_center;
+		bool should_tiebreak = false;
+		if (*r_closest != nullptr) {
+			// FIXME: This could be removed in favor of checking r_closest directly further down.
```

Comment:
What would it take to solve this FIXME?
FIXMEs and TODOs tend to never be fixed or done once the PR is merged.

Reply:
We would remove the `should_tiebreak` bool and inline the check directly further down, on line 2403:
```diff
-   } else if (should_tiebreak && d == r_closest_dist) {
+   } else if ((*r_closest != nullptr) && d == r_closest_dist) {
```
I was not sure what was preferred between setting a named boolean flag or doing the check directly, so I went with a boolean because it makes the code more readable in the context of a review.

Reply:
If you are unsure about some part, it's preferable to leave a comment in the PR itself, not in code.

Reply:
I just removed the FIXME. In my opinion the current explicit boolean flag is more readable than the alternative of checking `*r_closest != nullptr` directly.